job-server/src/spark.jobserver/auth/LdapGroupRealm.scala

```diff
@@ -0,0 +1,144 @@
+package spark.jobserver.auth
+
+import org.apache.shiro.realm.ldap.JndiLdapRealm
+import org.apache.shiro.realm.ldap.LdapContextFactory
+import org.apache.shiro.realm.ldap.JndiLdapContextFactory
+import org.apache.shiro.realm.ldap.LdapUtils
+import org.apache.shiro.subject.PrincipalCollection
+import org.apache.shiro.authz._
+import javax.naming.ldap.LdapContext
+import javax.naming.directory._
+import javax.naming.NamingEnumeration
+import org.slf4j.LoggerFactory
+import com.typesafe.config.{ Config, ConfigFactory }
+
+/**
+ * LDAP realm implementation that retrieves group information from LDAP and matches
+ * the 'member' attribute values of each group against the given user.
+ *
+ * @note not all LDAP installations use the member property.... we might have to add
+ *   memberOf matches as well and others
+ *
+ * @author KNIME (basics in Java stem from different sources by various authors from stackoverflow and such)
+ */
+class LdapGroupRealm extends JndiLdapRealm {
+  import collection.JavaConverters._
+
+  private val logger = LoggerFactory.getLogger(getClass)
+
+  private val searchCtls: SearchControls = {
+    val c = new SearchControls();
+    c.setSearchScope(SearchControls.SUBTREE_SCOPE);
+    c
+  }
+
+  lazy val searchBase: String = getContextFactory() match {
+    case jni: JndiLdapContextFactory =>
+      getEnvironmentParam(jni, "ldap.searchBase")
+    case _ =>
+      throw new RuntimeException("Configuration error: " +
+        "LdapGroupRealm requires setting of the parameter 'searchBase'")
+  }
+
+  lazy val allowedGroups: Option[Array[String]] = getContextFactory() match {
+    case jni: JndiLdapContextFactory =>
+      val groups = getEnvironmentParam(jni, "ldap.allowedGroups").
+        split("\"").filter(group => group.replaceAll("[, ]", "").length() > 0)

```

Comment:
nit: since " is just one char you can do '"' (... but I'm not sure if it will be more readable)